### PR TITLE
Fix deprecation warnings

### PR DESCRIPTION
**Pull request by msimberg**
Fixes more deprecation warnings and enables deprecation warnings on all the Jenkins builders. The main problem was that clang does not emit warnings for template aliases, whereas GCC does, and we only had deprecation warnings turned on with clang. GCC 10 also has a bug where using typedefs in deprecated functions triggers a deprecation warning (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=95302) so I've worked around that.

### CODE REVIEW

File: libs/full/compute_cuda/examples/hello_compute.cu [hunk 1]
```diff
@@ -48,40 +48,37 @@ int hpx_main(hpx::program_options::variables_map& vm)
     hpx::compute::vector<int, allocator_type> d_C(N, alloc);
 
     hpx::parallel::copy(
-        hpx::parallel::execution::seq,
-        h_A.begin(), h_A.end(), d_A.begin());
+        hpx::parallel::execution::seq, h_A.begin(), h_A.end(), d_A.begin());
```
Comment (hkaiser) [inline libs/full/compute_cuda/examples/hello_compute.cu:51]: Shouldn't this be `hpx::execution::seq`?

File: libs/full/compute_cuda/tests/unit/default_executor.cu [hunk 2]
```diff
@@ -67,7 +67,7 @@ void test_bulk_sync()
 
     hpx::cuda::experimental::target target;
     executor exec(target);
-//    traits::bulk_execute(exec, hpx::util::bind(&bulk_test, _1), v);
+    //    traits::bulk_execute(exec, hpx::util::bind(&bulk_test, _1), v);
```
Comment (hkaiser) [inline libs/full/compute_cuda/tests/unit/default_executor.cu:70]: Let's remove this while we're at it...

File: libs/full/compute_cuda/tests/unit/for_loop_compute.cu [hunk 3]
```diff
@@ -25,33 +25,26 @@ typedef hpx::cuda::experimental::default_executor executor_type;
...
     // FIXME : Lambda function given to for_loop_n() is momentarily defined as
     //         HPX_HOST_DEVICE in place of HPX_DEVICE to allow the host_side
     //         result_of<> (used inside for_each()) to get the return
     //         type
 
-    hpx::for_loop_n(
-        hpx::parallel::execution::par.on(exec),
-        d_A.data(), d_A.size(),
-        hpx::parallel::induction(d_B.data()),
+    hpx::for_loop_n(hpx::parallel::execution::par.on(exec), d_A.data(),
```
Comment (hkaiser) [inline libs/full/compute_cuda/tests/unit/for_loop_compute.cu:36]: Same here: `hpx::execution::par`?

File: libs/full/compute_cuda/examples/hello_compute.cu (hunk 1 above)
Comment (msimberg) [reply libs/full/compute_cuda/examples/hello_compute.cu:51]: It should :/ I suppose the flags don't get applied for nvcc... Will check if we can enable them. I'll fix this in any case. Very good catch!

**Review by hkaiser — COMMENTED** [review]
Thanks for working on this!

File: libs/full/compute_cuda/examples/hello_compute.cu (hunk 1 above)
Comment (msimberg) [reply libs/full/compute_cuda/examples/hello_compute.cu:51]: Actually they'll most likely trigger warnings. I just didn't test this with CUDA enabled.

**Review by aurianer — DISMISSED** [review]
Thanks!

File: libs/full/compute_cuda/examples/data_copy.cu [hunk 4]
```diff
@@ -45,11 +45,11 @@ int hpx_main(hpx::program_options::variables_map& vm)
 
     // copy data from host to device
     hpx::parallel::copy(
```
Comment (aurianer) [inline libs/full/compute_cuda/examples/data_copy.cu:47]: Isn't this supposed to be `hpx::copy` now?

Comment (msimberg) [reply libs/full/compute_cuda/examples/data_copy.cu:47]: Yep, good point. I still have to do a round of checks on the cuda builds.

**Comment by msimberg** [comment]
I've combined this with #4993 since there was quite some overlap between the two PRs. Closing this one.

**Pull request closed without merging**